### PR TITLE
CUMULUS-657

### PR DESCRIPTION
**Pull request by markdboyd**
* Adds test to verify that a rule can be deleted from the rules list page and that after deletion the rule no longer appears on the page

### CODE REVIEW

**Comment by markdboyd** [comment]
I didn't add any assertion/test for `When not logged in, attempting to delete a rule fails` because I'm not sure what that would mean in this context. We already have a test verifying that you can't view the rules listing page if you aren't logged in and since the Delete button appears on the listing page, it would seem unnecessary to add another test for this case.

File: cypress/integration/rules.spec.js [hunk 1]
```diff
@@ -61,5 +61,27 @@ describe('Rules page', () => {
...
+      cy.get('.form--controls button')
+        .contains('Delete')
+        .should('exist')
+        .click();
+      cy.get('.modal')
+        .should('exist')
+        .get('button')
+        .contains('Confirm')
+        .click();
+      cy.get('table tbody tr').should('not.exist');
```
Comment (jennyhliu) [inline cypress/integration/rules.spec.js]: This won't work if we have multiple rules.

File: test/fake-api-db/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return data;
+      })
+      .then((data) => seed(this.filePath, data));
+  }
+
+  getItems () {
+    return fs.readJson(this.filePath);
+  }
+
+  deleteItem (name) {
```
Comment (jennyhliu) [inline test/fake-api-db/index.js]: This only works for `rule`

Comment (markdboyd) [reply test/fake-api-db/index.js]: Yeah, I thought that storage for other types of data could override. Would you prefer I move this to the `FakeRulesDb` class?

Comment (jennyhliu) [reply test/fake-api-db/index.js]: yes.  delete collection for example has `name and version` as argument.

Comment (markdboyd) [reply test/fake-api-db/index.js]: OK. I moved it to the `FakeRulesDb` class

**Pull request merged**